src/queue.h

```diff
@@ -254,6 +254,93 @@ struct {								\
 	_Q_INVALIDATE((elm)->field.le_next);				\
 } while (0)
 
+/*
+ * Singly-linked Tail queue declarations.
+ */
+#define STAILQ_HEAD(name, type)                                                                    \
```

Comment:
I have already shared in the first request, to check if DPDK already has these definitions or not. Because Internally DPDK libraries use these. Please rework

Reply:
I am aware that DPDK libraries use these definitions internally as pointed in [the earlier comment](https://github.com/OISF/suricata/pull/6269#discussion_r671025820). However, to my knowledge, thanks to the way Suricata redefines `sys/queue.h`, you can only add the required macros to the `src/queue.h` file.